lib/spack/spack/ci.py

```diff
@@ -1973,9 +1989,10 @@ def reproduce_ci_job(url, work_dir, autostart, gpg_url, runtime):
                 "-v",
                 ":".join([os.path.join(work_dir, "spack"), mount_as_dir, "Z"]),
                 "--entrypoint",
-                os.path.join(mounted_workdir, "entrypoint.sh"),
+                entrypoint,
                 job_image,
-                "bash",
+                script,
```

Comment:
As a general note, IMO the pattern of composing a list from [x, y, z] where each component may or may not be present depending on variables (e.g. platform) can be hard to reason about. In this particular case, I don't see `script` referenced elsewhere, so I think it would be clearer to expand this particular case to `entry_path if IS_WINDOWS else ""`

Also I'm trying to understand the point of having both `--entrypoint` and arguments after the image (this is partially a concern with how things already were, but since we are modifying this code it might be useful to clean it up):

* On Windows, with this PR, you get `--entrypoint powershell.exe <image> script.ps1 powershell.exe`
* On Linux, before this PR, you get `--entrypoint .../entrypoint.sh <image> bash`
  * If `entrypoint.sh` didn't expect any arguments, this probably works fine (the extra "bash" is unnecessary, but would not cause problems)
  * On Linux, with this PR, it looks like you get the same thing

Reply:
Heard w.r.t. `script`.

W.r.t. `--entrypoint`
The previous (if interactive is true) behavior was to write `exec $@` as the final command of the entrypoint script. `bash` was then passed to it, as `--entrypoint ../entrypoint.sh <image> bash` so the user is dropped into an interactive bash with the setup Spack environment in the container.

The powershell version is the equivalent of this behavior, the only difference is that we need powershell to drive the script since we have no shebangs to leverage, so the script comes after the image where the entrypoint arguments go.